### PR TITLE
feat: add launch in fullscreen on startup toggle

### LINKED ISSUE #18: Default Fullscreen Mode
There should be a way of setting the display window to fullscreen by default, hiding the menu. This is useful in the situation that core/desktop is installed on a laptop, and you just want to control it from a phone or laptop straight away.

### PR DESCRIPTION
**Pull request by AkalUstat**
when the main window is ready-to-show. Toggling this does not affect while the app is running

### Summary of PR
On startup, if the user has the `fullscreen` setting enabled, the main window will open in fullscreen. Defaults to `true`

### Tests for unexpected behavior
Currently testing: when this is enabled, everything opens in fullscreen (settings, etc). Not sure if this is wanted behavior

### Time spent on PR
more time than needs to be known

### Linked issues
Fix #18 

### Reviewers
@bhajneet @Harjot1Singh

### CODE REVIEW

**Comment by bhajneet** [comment]
I know you're still working on this, no worries. Just a small note once you're done with the draft: fullscreen option should be off by default.

**Comment by AkalUstat** [comment]
Can't get this working as you want (just the display windows)...I spent like 20-25 min actually setting up and now upwards of 4hrs debugging and its not going anywhere.

**Comment by Harjot1Singh** [comment]
Leave the PR open and mark as draft, we can assist

**Comment by Harjot1Singh** [comment]
Also, rename the `fullscreen` key to something a little more descriptive, like`fullscreenOnLaunch`

**Comment by AkalUstat** [comment]
still have to push some changes i was playing with

**Comment by AkalUstat** [comment]
@Harjot1Singh cant find a solution to this

**Comment by bhajneet** [comment]
Is this actively being worked on? Please update the linked issue if so

**Comment by AkalUstat** [comment]
yea im playing around with some items. Turns out this applies everything gets fullscreened only applies to mac (since it fullscreens all windows of an app) -> might be a workaround for that. Windows is fine

**Comment by bhajneet** [comment]
@AkalUstat Please accept the triage permission for desktop repo (you have to be logged in before clicking the link in your email). This will let you change the issue's status label from "to do" to "wip"

**Comment by AkalUstat** [comment]
@bhajneet the Mac workaround doesnt work  very well. Since the fullscreen on each tab of the app is a mac default setting that each app has (and this issue doesnt appear on windows), im leaving it as it and opening up the pr

**Comment by bhajneet** [comment]
@Harjot1Singh consider dropping support for mac on this option? Only make it available to windows users?

**Comment by Harjot1Singh** [comment]
Not sure. I think the issue is more around the semantics of fullscreening on Mac generally, as opposed to this particular feature, so I would suggest merging it in, but think about/investigate how our app actually works in terms of fullscreening on Mac.

File: app/electron/electron-wrapper.js [hunk 1]
```diff
@@ -17,6 +17,12 @@ app.on( 'ready', () => {
 } )
 
 const onSettingsChange = ( { system } ) => {
+  // Fullscreen any display windows on start, if set
+  const windows = getDisplayWindows()
+  Object.values( windows ).forEach( window => window.once( 'ready-to-show', () => {
+    window.setFullScreen( system.fullscreen )
```
Comment (saihaj) [inline app/electron/electron-wrapper.js:23]: `system.fullscreen` is undefined. It should be `window.setFullScreen( system.fullscreenOnLaunch )`

**Comment by saihaj** [comment]
Doing this also makes all other items fullscreen too. Like if you open settings (and this settings is enabled) it is in fullscreen mode. Is this expected?

**Comment by AkalUstat** [comment]
yes this is the mac default...windows does not expereince this

File: app/electron/electron-wrapper.js (hunk 1 above)
Comment (saihaj) [inline app/electron/electron-wrapper.js:23]:
```suggestion
    window.setFullScreen( system.fullscreenOnLaunch )
```

**Comment by Harjot1Singh** [comment]
Using simple fullscreen on mac produces the desired behaviour. @saihaj please confirm this works as expected now.

**Comment by saihaj** [comment]
Yep, that works. Do we not want it full screen by default?

**Comment by Harjot1Singh** [comment]
We do not want it to fullscreen by default

**Pull request merged**